src/subscriptions.c

```diff
@@ -906,14 +911,18 @@ subscription_status_callback ( void *p )
              subscription_status_callback, NULL, 1);
 
   LIST_FOREACH(s, &subscriptions, ths_global_link) {
-    int errors  = s->ths_total_err;
-    int in      = atomic_exchange(&s->ths_bytes_in, 0);
-    int out     = atomic_exchange(&s->ths_bytes_out, 0);
+    /* Store the difference between total bytes from the last round */
+    uint64_t in_prev = s->ths_total_bytes_in_prev;
+    uint64_t in_curr = s->ths_total_bytes_in;
+    uint64_t out_prev = s->ths_total_bytes_out_prev;
+    uint64_t out_curr = s->ths_total_bytes_out;
+
+    atomic_exchange(&s->ths_bytes_in_avg, (in_curr - in_prev));
+    atomic_exchange_u64(&s->ths_total_bytes_in_prev, s->ths_total_bytes_in);
+    atomic_exchange(&s->ths_bytes_out_avg, (out_curr - out_prev));
+    atomic_exchange_u64(&s->ths_total_bytes_out_prev, s->ths_total_bytes_out);
```

Comment:
All four atomic operations here are not required. The access is already protected using "global_lock".

The only "missing" protection is above - the read operation should be protected (it's probably not an issue for x86, but for other CPUs, the 64-bit operations might be more complex):

<pre>
int64_t in_curr = atomic_add(&s->ths_total_bytes_in, 0);
</pre>